test/jdk/jdk/incubator/vector/AbstractVectorConversionTest.java

```diff
@@ -511,4 +511,50 @@ static void copyConversionArray(Object src, int srcPos,
 
         Assert.assertEquals(actual, expected);
     }
+
+    static <E,F> void legal_mask_cast_kernel(VectorSpecies<E> src, VectorSpecies<F> dst) {
+        for(int i = 0; i < INVOC_COUNT; i++) {
+            VectorMask<E> mask = VectorMask.fromLong(src, i);
+            VectorMask<F> res = mask.cast(dst);
+            Assert.assertEquals(res.toLong(), mask.toLong());
+        }
+    }
+
+    static <E,F> void illegal_mask_cast_kernel(VectorSpecies<E> src, VectorSpecies<F> dst) {
+        for(int i = 0; i < INVOC_COUNT; i++) {
```

Comment:
We don't need the `INVOC_COUNT` loop for the illegal case, since the checks are performed in Java and not in the intrinsic.